### PR TITLE
add subscribe event filter and updated tests

### PR DESCRIPTION
**Pull request by dsawali**
#1746 

Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have added unit tests
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: packages/taquito/src/subscribe/filters.ts [hunk 1]
```diff
@@ -50,6 +52,26 @@ const destinationFilter = (x: OperationContent, filter: DestinationFilter) => {
...
+      }
+    } else if (!address) {
+      if (tag) {
+        return result.tag === tag;
+      } else if (!tag) {
+        return true;
+      }
+    }
+  } else {
+    return false;
```
Comment (roxaneletourneau) [inline packages/taquito/src/subscribe/filters.ts:71]: Could you add one more unit test to cover this line?

File: integration-tests/polling-subscribe-provider.spec.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import { CONFIGS } from "./config";
+import { Protocols, PollingSubscribeProvider } from "@taquito/taquito";
+
+const eventContractAddress = 'KT1GXjpeB63bXGEoMB9HR5mj6fPS7rfWSjqW';
```
[the remark this answers is not in the document]
Comment (dsawali) [reply integration-tests/polling-subscribe-provider.spec.ts:4]: very good point, I'll update that

File: integration-tests/polling-subscribe-provider.spec.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+      await sleep(5000);
+
+      eventSub.close();
+
+      expect(data.length).toEqual(1);
+      expect(data[0].opHash).toBeDefined();
+      expect(data[0].blockHash).toBeDefined();
+      expect(data[0].level).toBeDefined();
+      expect(data[0].kind).toEqual('event');
```
[the remark this answers is not in the document]
Comment (dsawali) [reply integration-tests/polling-subscribe-provider.spec.ts:46]: that was included in the internal_operation_result response from the chain, we can remove it if it seems redundant

[the remark this answers is not in the document]
Comment (roxaneletourneau) [reply integration-tests/polling-subscribe-provider.spec.ts:46]: It is not a big deal, up to you to keep it or not.

File: docs/events.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+### Example
+#### Usage
+```typescript
+const Tezos = new TezosToolkit(RPC_URL);
+
+Tezos.setStreamProvider(
+  Tezos.getFactory(PollingSubscribeProvider)({
+    shouldObservableSubscriptionRetry: true, 
+    pollingIntervalMilliseconds: 15000
```
Comment (roxaneletourneau) [inline docs/events.md:25]: A polling interval of 15000 is high and might leads to skipping some blocks. 
By default, the interval will be set using the RPC getConstants. We don't necessarily need to override it in the example.

**Review by roxaneletourneau — APPROVED** [review]
lgtm :)

**Pull request merged**